trace-sdk/src/main/java/io/bitrise/trace/configuration/ConfigurationManager.java

```diff
@@ -257,17 +257,27 @@ public boolean isAppDebugBuild() {
    * @return the DataCollectors.
    */
   @NonNull
-  public Set<DataCollector> getDataCollectors(@NonNull final Context context) {
+  public Set<DataCollector> getRecurringDataCollectors(@NonNull final Context context) {
     final Set<DataCollector> dataCollectors = new HashSet<>();
     dataCollectors.add(new ApplicationUsedMemoryDataCollector(context));
     dataCollectors.add(new SystemMemoryDataCollector(context));
-
     dataCollectors.add(new SystemCpuUsageDataCollector());
     dataCollectors.add(new ApplicationCpuUsageDataCollector());
+    return dataCollectors;
+  }
 
+  /**
+   * Creates a new set of {@link DataCollector}s that only need to collect their data once during
+   * the lifecycle of the application e.g. Application version codes.
+   *
+   * @param context the Android Context.
+   * @return the DataCollectors.
+   */
+  @NonNull
+  public Set<DataCollector> getSingleDataCollectors(@NonNull final Context context) {
```

Comment:
i didn't have a better name for "something that needs to collect data only once" :/ Do you have any other suggestion?